x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/data/DoubleBigArrayVector.java

```diff
@@ -60,7 +60,11 @@ public long ramBytesUsed() {
 
     @Override
     public DoubleVector filter(int... positions) {
-        return new FilterDoubleVector(this, positions);
+        final DoubleArray filtered = blockFactory.bigArrays().newDoubleArray(positions.length, true);
+        for (int i = 0; i < positions.length; i++) {
+            filtered.set(i, values.get(positions[i]));
+        }
+        return new DoubleBigArrayVector(filtered, positions.length, blockFactory);
```

Comment:
I think we don't account the memory usage of BigArrayVector correctly. However, we aren't using them now. I can fix them in a follow-up.

Reply:
+++

Reply:
++